### ISSUE TITLE
Add F1 Score metric

### ISSUE THREAD

**Issue by jshin1394** [google/metrax#36]

**Comment by jshin1394** [comment]
Let's start by investigating this [Keras implementation of F1Score](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/F1Score).

**Comment by nikolasavic3** [comment]
In Keras the [F1Score](https://github.com/keras-team/keras/blob/f6c4ac55692c132cd16211f4877fac6dbeead749/keras/src/metrics/f_score_metrics.py#L250-L320) is subclassing the [FBetaScore](https://github.com/keras-team/keras/blob/f6c4ac55692c132cd16211f4877fac6dbeead749/keras/src/metrics/f_score_metrics.py#L9-L248). Should we implement it that way here to?

**Comment by jshin1394** [comment]
I think it is a good idea to support both FBetaScore and F1Score unless it makes the implementation a lot more complicated.

**Comment by jeffcarp** [comment]
Assigning to @nikolasavic3 for bookkeeping - feel free to unassign based on your bandwidth (no problem either way)

**Comment by Heidar-An** [comment]
I'll work on this if not completed by June 10th!

**Comment by 2468cool** [comment]
I guess I'm a little late to mention anything but I've been working on this problem as well. I worked on adding F-Beta score metric as best I can where beta defaults to 1 since F1 score formula is the F-Beta score when beta = 1. Am I still good to make a pull request for my work?

**Comment by jshin1394** [comment]
Hey @Heidar-An, please let us know if you are occupied with other metrics at this time :)

**Comment by 2468cool** [comment]
I think I'm getting pretty close. Right now only the empty test and the batch size one test for F-Beta pass. For the ones that don't pass it seems that the keras F-Beta gives an array of values while mine currently only gives one value regardless. Keras's F-Beta doesn't like the array reshaped into a one dimensional array so I made the test case reshaping of y_true and y_pred only affect the metrax F-Beta I made. Leaving everything as is without reshaping produces the same results of only the batch size 1 passing.

**Comment by 2468cool** [comment]
Quick update:
when doing the tests for F-Beta. I saw that the y_true and y_pred are modified as such:

y_true = y_true.reshape((-1,))
y_pred = jnp.where(y_pred.reshape((-1,)) >= threshold, 1, 0)

Keras's FBeta class does NOT like this, changing the modification to:

y_true = y_true.reshape((-1, 1))
y_pred = jnp.where(y_pred.reshape((-1, 1)) >= threshold, 1, 0)

makes it so Keras's FBeta can take them in. Now only 5 test cases are failing.

**Comment by 2468cool** [comment]
So for the recall and precision tests the data is outputted as a single number for both Keras and Metrax versions. For Keras's F-Score metric, depending on how the data is fed into it, can give several outputs for the input. Should Metrax F-Score also do this or should it only be a single number output like precision and recall? My current set-up is like the latter.

**Comment by jshin1394** [comment]
Hi Benjamin, thank you so much for your work. Would you like to share your implementation with me via a PR? I think it would be easier to move the conversation there :).

**Comment by 2468cool** [comment]
Should we close this issue?

**Comment by jshin1394** [comment]
Yes :)
